File: examples/python/scan_barcode.py
"""
CalsCall Nutrition API — Python example

Install:
    pip install requests

Usage:
    python scan_barcode.py 5053990194769
"""

import sys
import requests

API_KEY = "your-key-here"
BASE_URL = "https://calscall.online"


def lookup_barcode(barcode: str) -> dict:
    headers = {"X-API-Key": API_KEY}
    resp = requests.get(
        f"{BASE_URL}/v1/nutrition",
        params={"barcode": barcode},
        headers=headers,
        timeout=10,
    )
    resp.raise_for_status()
    return resp.json()


if __name__ == "__main__":
    barcode = sys.argv[1] if len(sys.argv) > 1 else "5053990194769"
    data = lookup_barcode(barcode)

    if data["found"]:
        p = data["p"]
        print(f"✅ {p['n']}")
        print(f"   Calories:  {p['c']} kcal/100g")
        print(f"   Protein:   {p['p']}g/100g")
        print(f"   Carbs:     {p['h']}g/100g")
        print(f"   Fat:       {p['f']}g/100g")
        if data.get("sv"):
            print(f"   Per serve: {data['sv']['c']} kcal")
        print(f"   Source:    {data['source']}")
        if p["ns"]:
            print(f"   NutriScore: {p['ns']}")
        if p["al"]:
            print(f"   Allergens:  {p['al']}")
    else:
        print(f"❌ Product not found for barcode {barcode}")

File: examples/nodejs/scan_barcode.mjs
/**
 * CalsCall Nutrition API — Node.js example
 *
 * Usage:
 *   node scan_barcode.mjs 5053990194769
 */

const API_KEY = "your-key-here";
const BASE_URL = "https://calscall.online";

async function lookupBarcode(barcode) {
  const resp = await fetch(
    `${BASE_URL}/v1/nutrition?barcode=${encodeURIComponent(barcode)}`,
    { headers: { "X-API-Key": API_KEY } }
  );
  if (!resp.ok) throw new Error(`HTTP ${resp.status}: ${await resp.text()}`);
  return resp.json();
}

const barcode = process.argv[2] || "5053990194769";

lookupBarcode(barcode)
  .then((data) => {
    if (data.found) {
      const p = data.p;
      console.log(`✅ ${p.n}`);
      console.log(`   Calories:  ${p.c} kcal/100g`);
      console.log(`   Protein:   ${p.p}g/100g`);
      console.log(`   Carbs:     ${p.h}g/100g`);
      console.log(`   Fat:       ${p.f}g/100g`);
      if (data.sv) console.log(`   Per serve: ${data.sv.c} kcal`);
      console.log(`   Source:    ${data.source}`);
      if (p.ns) console.log(`   NutriScore: ${p.ns}`);
      if (p.al) console.log(`   Allergens:  ${p.al}`);
    } else {
      console.log(`❌ Product not found for barcode ${barcode}`);
    }
  })
  .catch(console.error);
